cli/daemon/db.go

```diff
@@ -20,6 +23,18 @@ import (
 
 // DBConnect starts the database and returns the DSN for connecting to it.
 func (s *Server) DBConnect(ctx context.Context, req *daemonpb.DBConnectRequest) (*daemonpb.DBConnectResponse, error) {
+	serviceExist := false
+	filepath.WalkDir(req.AppRoot, func(path string, info fs.DirEntry, err error) error {
+		if info.IsDir() && filepath.Base(path) == req.DbName {
+			serviceExist = true
+			return io.EOF
+		}
+		return nil
+	})
+	if !serviceExist {
+		return nil, errServiceNotFound
+	}
```

Comment:
As I mentioned before we should not look at just the directory names but inspect the Encore application metadata. 

Move this down into `dbConnectLocal` and instead of walking the filesystem we should inspect the `parse.Meta` object. It is a protobuf-generated data structure that you can see here: https://github.com/encoredev/encore/blob/main/proto/encore/parser/meta/v1/meta.proto#L10

What we need to do is to iterate over the `Svcs` list and see if one exists that matches the `req.DbName`.

If so, we know that the service exists, and then we need to check whether that service has a database.

The way to do that is by checking if the `Migrations` slice is non-empty, see: https://github.com/encoredev/encore/blob/main/proto/encore/parser/meta/v1/meta.proto#L46